packages/web3-utils/src/converters.ts

```diff
@@ -173,12 +174,17 @@ export const toDecimal = hexToNumber;
  * > "0xa"
  * ```
  */
-export const numberToHex = (value: Numbers): HexString => {
+export const numberToHex = (value: Numbers, hexstrict?: boolean): HexString => {
```

Comment:
Added an optional parameter hexstrict,

if a user tries to do `HexToBytes(numberToHex(3))` this will fail do to `0x3` not being hexstrict, 

numberToHex(3, true) will return `0x03` allowing `HexToBytes(numberToHex(3, true))`  to work